cli/client.go

```diff
@@ -593,15 +595,19 @@ var clientRetrieveCmd = &cli.Command{
 			return fmt.Errorf("The received offer errored: %s", offer.Err)
 		}
 
+		maxPrice := types.NewInt(DefaultMaxRetrievePrice)
+
 		if cctx.String("maxPrice") != "" {
-			maxPrice, err := types.ParseFIL(cctx.String("maxPrice"))
+			maxPriceFil, err := types.ParseFIL(cctx.String("maxPrice"))
 			if err != nil {
-				return err
+				return xerrors.Errorf("parsing maxPrice: %w", err)
 			}
 
-			if offer.MinPrice.GreaterThan(types.BigInt(maxPrice)) {
-				return xerrors.Errorf("Failed to find offer satisfying maxPrice: %w", maxPrice)
-			}
+			maxPrice = types.BigInt(maxPriceFil)
```

Comment:
Should be
```suggestion
			maxPrice = types.FromFil(maxPriceFil)
```

Otherwise this will set the max to 1 attoFil

Reply:
Wait, meant here - https://github.com/filecoin-project/lotus/pull/2347/files#diff-b47a2d7fe9543751ffd6d9e6a5be7433R598